tests/AbstractTestCase.php

```diff
@@ -35,10 +30,72 @@
      */
     public function createApplication()
     {
-        $app = require __DIR__.'/../bootstrap/app.php';
+        $app = require __DIR__ . '/../bootstrap/app.php';
 
         $app->make(Kernel::class)->bootstrap();
 
         return $app;
     }
+
+    /**
+     * Sign in an user if it's the case.
+     *
+     * @param User|null $user
+     *
+     * @return TestCase
+     */
+    protected function signIn($user = null)
+    {
+        if (isset($user)) {
+            $this->user = $user;
+        }
+
+        if (is_null($this->user)) {
```

Comment:
There is no need to check `$this->user` when it's equivalent to `$user`, so you can make the code this:

```php
if ($user) {
    $this->user = $user;
} else {
    $this->user = $this->createUser();
}
```